databases/business-sync/fragment-shoebox-snippet.js

```diff
@@ -0,0 +1,40 @@
+{
+  channels: getDocSyncChannels(doc, oldDoc),
+  typeFilter: function(doc, oldDoc) {
+    return createBusinessEntityRegex('shoeboxSnippet\\.[A-Za-z0-9_-]+\\.[A-Za-z0-9_-]+').test(doc._id);
+  },
+  allowUnknownProperties: true,
+  immutable: true,
+  propertyValidators: {
+    type: {
+      // The type of shoebox snippet (ie, bank transaction, email, photo, digital receipt)
+      type: 'string',
+      required: true,
+      mustNotBeEmpty: true
+    },
```

Comment:
Personally I like to have the field available on the entity rather than needed to deduce from the key (at least when programming against the entity).  I think it also gives more options for querying (ie, a N1QL query with an index on that field may be faster than applying a regex across all keys.  Or maybe not, but at least we have both options to weigh against each other)